dll/win32/shell32/wine/control.c

```diff
@@ -1173,20 +1173,70 @@ static	void	Control_DoLaunch(CPanel* panel, HWND hWnd, LPCWSTR wszCmd)
 void WINAPI Control_RunDLLW(HWND hWnd, HINSTANCE hInst, LPCWSTR cmd, DWORD nCmdShow)
 {
     CPanel	panel;
+#ifdef __REACTOS__
+    WCHAR *ptr1 = NULL;
+    WCHAR *ptr2 = NULL;
+    WCHAR comma = ',';
+    WCHAR dot = '.';
+    LPWSTR buffer;
+    WCHAR ext[4] = L"   ";
+    SIZE_T nLen = lstrlenW(cmd);
+    BOOL GoodExt;
+
+    TRACE("(%p, %p, %s, 0x%08x)\n",
+          hWnd, hInst, debugstr_w(cmd), nCmdShow);
+
+/* Search 'cmd' for first comma. If the remaining length is 1 or 2,
+ * then replace dot before extention with UNICODE_NULL. */
+
+    buffer = HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, sizeof(*buffer) * (nLen + 1));
+    if (!buffer) return;
+
+    lstrcpyW(buffer, cmd);
+
+    ptr1 = wcschr(cmd, comma);
+    if (ptr1)
+        nLen = lstrlenW(ptr1+1);
+    else
+        nLen = 0;
+
+    ptr2 = wcschr(cmd, dot);
```

Comment:
This seems fairly brittle, something with multiple dots in the name wont work here.

Reply:
Would you like to propose a sequence of commands that work on W2K3SP2 and fail here that use multiple dots?
I will be happy to test any that you think might be affected in this way. I cannot find any myself at this time.

Reply:
Just take a valid CPL file on Windows and rename it to something like `my.test.file.cpl`. Then run the Control_RunDLLW on it.

Reply:
New code now replaces the above with this:
```
ptr2 = PathFindExtensionW(cmd);
```
Thanks

Reply:
This solution finds the extension without regard to how many dots are in the file name.
I have tested this extensively to confirm this.